subgraphs/uniswap-v2/src/common/helpers.ts

```diff
@@ -0,0 +1,167 @@
+      financialMetrics.protocol = FACTORY_ADDRESS;
+
+      financialMetrics.feesUSD = BIGDECIMAL_ZERO
+      financialMetrics.totalVolumeUSD = BIGDECIMAL_ZERO
+      financialMetrics.totalValueLockedUSD = BIGDECIMAL_ZERO
+      financialMetrics.supplySideRevenueUSD = BIGDECIMAL_ZERO
+      financialMetrics.protocolSideRevenueUSD = BIGDECIMAL_ZERO
+  }
+  
+    // Update the block number and timestamp to that of the last transaction of that day
+  financialMetrics.blockNumber = blockNumber;
+  financialMetrics.timestamp = timestamp;
+  financialMetrics.totalValueLockedUSD = protocol.totalValueLockedUSD
+
+  financialMetrics.save();
+}
+
+
+
+export function updateUsageMetrics(blockNumber: BigInt, timestamp: BigInt, from: Address): void {
+  // Number of days since Unix epoch
+    let id: i64 = timestamp.toI64() / SECONDS_PER_DAY;
+    let usageMetrics = UsageMetricsDailySnapshot.load(id.toString());
+    let uniqueUsersTotal = _HelperStore.load("USERS")
+    if (uniqueUsersTotal == null) return
+
+    if (!usageMetrics) {
+        usageMetrics = new UsageMetricsDailySnapshot(id.toString());
+        usageMetrics.protocol = FACTORY_ADDRESS;
+
+        usageMetrics.activeUsers = 0;
+        usageMetrics.totalUniqueUsers = 0;
+        usageMetrics.dailyTransactionCount = 0;
+    }
+
+    let accountId = from.toHexString()
+    let account = _Account.load(accountId)
+    if (!account) {
+        account = new _Account(accountId);
+        account.save();
+
+        uniqueUsersTotal.valueInt = uniqueUsersTotal.valueInt + INT_ONE
+    } 
+
+    // Combine the id and the user address to generate a unique user id for the day
+    let dailyActiveAccountId = id.toString() + "-" + from.toHexString()
+    let dailyActiveAccount = _DailyActiveAccount.load(dailyActiveAccountId);
+    
+    if (!dailyActiveAccount) {
+        dailyActiveAccount = new _DailyActiveAccount(dailyActiveAccountId);
+        dailyActiveAccount.save();
+        usageMetrics.activeUsers += 1;
+    }
+
+    // Update the block number and timestamp to that of the last transaction of that day
+    usageMetrics.blockNumber = blockNumber;
+    usageMetrics.timestamp = timestamp;
+    usageMetrics.dailyTransactionCount += 1;
+    usageMetrics.totalUniqueUsers = uniqueUsersTotal.valueInt
+
+    usageMetrics.save();
+
+}
+
+export function updatePoolMetrics(blockNumber: BigInt, timestamp: BigInt, pool: LiquidityPool): void {
+    // Number of days since Unix epoch
+      let id: i64 = timestamp.toI64() / SECONDS_PER_DAY;
+      let poolMetrics = PoolDailySnapshot.load(pool.id.concat("-").concat(id.toString()));
+  
+      if (!poolMetrics) {
+        poolMetrics = new PoolDailySnapshot(pool.id.concat("-").concat(id.toString()));
+        poolMetrics.protocol = FACTORY_ADDRESS;
+        poolMetrics.pool = pool.id;
+        poolMetrics.rewardTokenEmissionsAmount = [BIGDECIMAL_ZERO, BIGDECIMAL_ZERO]
+        poolMetrics.rewardTokenEmissionsUSD = [BIGDECIMAL_ZERO, BIGDECIMAL_ZERO]
```

Comment:
Same here

Reply:
Probably a good idea to add a comment in the schema along the lines of `Uniswap does not have reward tokens so these field will always be empty`